src/stores/set_problems.ts

```diff
@@ -106,9 +106,16 @@ export const useSetProblemStore = defineStore('set_problems', {
 		 * Fetch all set problems with given course_id.
 		 */
 		async fetchSetProblems(course_id: number): Promise<void> {
-			const response = await api.get(`courses/${course_id}/problems`);
-			const all_problems = response.data as Array<ParseableProblem>;
-			this.set_problems = all_problems.map((prob) => parseProblem(prob, 'Set') as SetProblem);
+			const response = await api.get(`courses/${course_id}/problems`)
+				.then((response) => {
+					if (response.status !== 200) throw (response.data as Error);
+					return response.data as Array<ParseableProblem>;
+				})
+				.catch((e: Error) => {
+					logger.error(`[fetchSetProblems] ${e.message}`);
+					throw e;
```

Comment:
This is really bad.  You are re-throwing the exception that you just caught.  Furthermore when the fetchSetProblems method is called, there is no exception handling.  Any exception that is thrown needs to be handled somewhere higher up in the call sequence, and re-throwing a caught exception is in general a bad idea.

Reply:
Yeah, this was an accidental inclusion -- it should have been pulled out. There were authentication issues with the stores testing and I inserted this in here to try and capture some info about what was going on. I'll update this PR to remove the catch altogether, see below.

More generally, @pstaabp and I discussed using axios hooks to throw exceptions when responses are not 200 OK. Then we'll need to go through the components and handle exceptions when dealing with store fetches... see #115. Is that along the lines of what you're thinking?

Reply:
Yeah, that TODO item is what I am thinking!